### DISCUSSION TITLE
[ Idea } Feature Request: Add Support for Cross-Seed Integration in Transmission Setup

### DISCUSSION THREAD

**Discussion by KnuffigerMaxi** [ahembree/ansible-hms-docker#106] [Ideas]
Hi,  
I’ve been using **ansible-hms-docker** to manage my media services and would like to propose enhanced support for integrating [[cross-seed](https://github.com/cross-seed/cross-seed)](https://github.com/cross-seed/cross-seed) with **Transmission**. Cross-seed is a tool that helps discover torrents from multiple trackers, improving redundancy and download speeds.

### Issue  
Currently, in my setup with **Transmission 4.0.5 (a6fe2a64aa)**, RPC 17, and WEB Control 1.6.1 (20200913), cross-seed organizes torrents into subdirectories based on tracker names, like this:

```bash
cat:
  movies-hd: /data/.torrents/movies/hd
  movies-uhd: /data/.torrents/movies/uhd
  cross-seed: /data/.torrents/movies/cross-seed
  HDB: /data/.torrents/movies/cross-seed/HDB/
  ATH: /data/.torrents/movies/cross-seed/ATH/
  MTV: /data/.torrents/movies/cross-seed/MTV/
  TL: /data/.torrents/movies/cross-seed/TL/
  BLU: /data/.torrents/movies/cross-seed/BLU/
```

However, cross-seed’s default category system and Transmission's behavior don’t align perfectly. Transmission lacks built-in support for tracker-specific categorization and instead organizes torrents broadly, which complicates integration.

### Proposed Solution  
1. **Customizable Category Mapping:** Add configuration options to allow Transmission to map cross-seed’s subdirectories (based on tracker names) to categories in Transmission.
2. **Directory Auto-Discovery:** Enable automatic scanning of `/data/torrents/movies/cross-seed` for new tracker subdirectories created by cross-seed.
3. **RPC Adjustments:** Improve RPC settings to sync with cross-seed’s torrent addition methods.

### Benefits  
- Simplifies torrent organization for cross-seed users in Transmission.  
- Enables better alignment with cross-seed’s tracker-based directory structure.  
- Streamlines automation for seedbox setups.

### References  
- [[Cross-seed GitHub Repository](https://github.com/cross-seed/cross-seed)](https://github.com/cross-seed/cross-seed)  
- [[Cross-seed Basics: Getting Started](https://www.cross-seed.org/docs/basics/getting-started)](https://www.cross-seed.org/docs/basics/getting-started)  
- [[Transmission RPC Protocol](https://github.com/transmission/transmission/wiki/RPC-Protocol)](https://github.com/transmission/transmission/wiki/RPC-Protocol)

**Comment by ahembree** [comment]
How will this impact torrents known to and categorized by Sonarr and/or Radarr? For example if Radarr adds its downloads to a "movies" category, will this work with cross-seed?

Seems like there's quite a bit of configuration outside of just running the container which isn't preferred

**Reply by KnuffigerMaxi** [reply]
Oh okay, too bad, it's really a good tool for private trackers as the .torrents are encrypted with a passkey. 
I don't have an answer to your question, I haven't looked into it enough.

**Comment by ahembree** [comment]
Not interested in adding this due to the work/effort involved and having to consider all edge cases.

If you get it working, feel free to submit a pull request.

**Reply by KnuffigerMaxi** [reply]
I have now simply integrated the cross-seed container as an additional container outside the stack. and this into hms-docker_proxy_net / hms-docker_download_net .... rather into proxy_net!
